src/hotspot/share/runtime/sharedRuntimeTrans.cpp

```diff
@@ -514,14 +514,16 @@ static double __ieee754_pow(double x, double y) {
   if(lx==0) {
     if(ix==0x7ff00000||ix==0||ix==0x3ff00000){
       z = ax;                   /*x is +-0,+-inf,+-1*/
-      if(hy<0) z = one/z;       /* z = (1/|x|) */
+      if(hy<0) {
+        if (z == 0.0) {
```

Comment:
Maybe s/z == 0.0/ix == 0/ ?

Reply:
Hi Kim, why ix and not x ? Is it a typo?

Reply:
Not a typo, I really meant `ix`.  The point is that testing the `int` typed `ix` for zero is no worse, and might
be better than, testing a `double` typed value for zero.  And I suggested `ix` rather than `hx` because `ix`
has the sign bit stripped off.